### DISCUSSION TITLE
Need help configuring for pinned vs. non-pinned NPM dependencies

### DISCUSSION THREAD

**Discussion by vladasmikalajunas** [renovatebot/renovate#31246] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### Please tell us more about your question or problem

I have a shared preset that defines package rule to create a single PR for all non-major NPM dependency upgrades. However, I want this rule to _exclude all pinned packages_. For these pinned packages, I would like Renovate to create separate PRs by following `config:recommended`.

Here’s the shared configuration preset (org/repo/allNonMajorNpm.json):
```json5
{
  "packageRules": [
    {
      "groupName": "all non-major NPM dependencies",
      "groupSlug": "all-minor-patch-npm",
      "matchManagers": [
        "npm"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "matchCurrentVersion": "!/^0/",
      "updatePinnedDependencies": false // this seems to be unsupported
    }
  ]
}
```

And in the project I have the following configuration:
```json5
{
  "extends": ["github>org/repo/allNonMajorNpm", "config:recommended"],
  "enabledManagers": ["npm"]
}
```

The `updatePinnedDependencies` option appears to only be supported globally at the root of the configuration, not within specific `packageRules`. As a result, I’m unable to configure Renovate to _exclude pinned packages from the grouped PR_.

One option I am considering is to extend `matchCurrentVersion` with a regex that would ignore package versions with flexible range characters (`^~*x`) but maybe there is a better option.

So, what is the best way to configure such behavior?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
To exclude pinned packages from the grouped PR try this:

``` "matchUpdatesTypes" : ["minor", "patch", "!pin"] ```

**Reply by vladasmikalajunas** [reply]
Just tried this and unfortunately it didn't work.

**Comment by rarkins** [comment]
You need to be using `matchCurrentValue`, not `matchCurrentVersion`

**Reply by vladasmikalajunas** [reply]
Oh, that's right. Thanks for noticing that.

**Reply by vladasmikalajunas** [reply]
So there is no better way to ignore pinned versions than something like `"matchCurrentValue": "!/^\d+\.\d+\.\d+/"`?

**Reply by rarkins** [reply, marked as answer]
That's the best way
